src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotificationsTrait.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.cloudbees.jenkins.plugins.bitbucket;
+
+import com.cloudbees.jenkins.plugins.bitbucket.hooks.WebhookConfiguration;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import jenkins.scm.api.SCMSource;
+import jenkins.scm.api.trait.SCMSourceContext;
+import jenkins.scm.api.trait.SCMSourceTrait;
+import jenkins.scm.api.trait.SCMSourceTraitDescriptor;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * A {@link SCMSourceTrait} for {@link BitbucketSCMSource} that sets the committersToIgnore
+ * setting in {@link WebhookConfiguration}.
+ *
+ * @since 2.10.0
+ */
+public class BitbucketBuildStatusNotificationsTrait extends SCMSourceTrait {
+
+    /**
+     * Should unstable builds be communicated as success to Bitbucket
+     */
+    @NonNull
+    private final boolean isUnstableBuildSuccess;
+
+    /**
+     * Constructor.
+     *
+     * @param isUnstableBuildSuccess should unstable builds be communicated as success to Bitbucket
+     */
+    @DataBoundConstructor
+    public BitbucketBuildStatusNotificationsTrait(@NonNull boolean isUnstableBuildSuccess) {
```

Comment:
If you only want this trait to control "Send Success notification for Unstable", that name should specifically say that. 

Otherwise, it should be more generic to allow for additional values later.  

Similarly, if you only plan to have one value having it be part of the constructor is fine.  If you think there might be additional values later, making the constructor empty and using a setter would be better. 


```suggestion
    public BitbucketBuildStatusNotificationsTrait() {
```

Reply:
Agreed, I did not consider that. I will make it more generic.

Reply:
added a setter in 30b253f